src/adapters/postgres/user-adapter.ts

```diff
@@ -387,7 +398,31 @@ export class PostgresUserService implements IServicelocator {
     }
   }
 
-  async validateBodyFields(userCreateDto) {
+  async validateRequestBody(userCreateDto, response, apiId) {
+    for (const [key, value] of Object.entries(userCreateDto)) {
+      if (key === 'email') {
+        const checkValidEmail = CustomFieldsValidation.validate('email', userCreateDto.email);
+        if (!checkValidEmail) {
+          return APIResponse.error(response, apiId, "BAD_REQUEST", `Invalid email address`, HttpStatus.BAD_REQUEST);
+        }
+      }
+
+      if (key === 'mobile') {
+        const checkValidEmail = CustomFieldsValidation.validate('mobile', userCreateDto.mobile);
+        if (!checkValidEmail) {
+          return APIResponse.error(response, apiId, "BAD_REQUEST", `Mobile number must be 10 digits long`, HttpStatus.BAD_REQUEST);
+        }
+      }
+
+      if (key === 'dob') {
+        const checkValidEmail = CustomFieldsValidation.validate('dob', userCreateDto.dob);
```

Comment:
const checkValidEmail = CustomFieldsValidation.validate('dob', userCreateDto.dob);
        this should be date